### PR TITLE
feature/AST-97

### PR DESCRIPTION
**Pull request by satyanarayanaRaju21**
+ list device type api

### CODE REVIEW

File: app.js
Comment (naveen-repos) [inline app.js:1]: I am expecting here we need to add cors config. Please check this by calling from another service and add If needed.

File: src/controllers/device/devices.constant.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+exports.successResponses = {
+  default: {
+    statusCode: 200,
+    message: '',
+  },
```
Comment (naveen-repos) [inline src/controllers/device/devices.constant.js:5]: get rid of the `default` here

File: src/controllers/device/devices.constant.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  COUNT_FETCH_SUCCESSFULL: 'data fetched successful',
+  NO_DATA_FOUND: 'no data found',
+  DATA_FETCH_SUCCESSFULL: 'data fetched successfully',
+};
+
+exports.errorResponses = {
+  default: {
```
Comment (naveen-repos) [inline src/controllers/device/devices.constant.js:12]: Keep this in to formate

File: src/utils/statusCode.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  STATUS_CODE_SUCCESS: 200,
+  STATUS_CODE_SUCCESS_CREATED: 201,
+  STATUS_CODE_INVALID_FORMAT: 400,
+  STATUS_CODE_UNAUTHORIZED: 401,
+  STATUS_CODE_FORBIDDEN: 403,
+  STATUS_CODE_DATA_NOT_FOUND: 404,
+  STATUS_CODE_FAILURE: 500,
+  STATUS_CODE_DATA_SOFT_DELETE: 421,
+  STATUS_CODE_VALIDATION_FAILED: 422,
+  STATUS_CODE_NOT_ACCEPTABLE: 406,
```
Comment (naveen-repos) [inline src/utils/statusCode.js:16]: Please place needed codes only

File: src/controllers/device/devices.query.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const { dbTables } = require('../../utils/constant');
+
+exports.queries = {
+  getDeviceCount: `SELECT
+  adv.device_type AS type,
+  COUNT(*) AS version_count,
+  (SELECT COUNT(*) FROM ${dbTables.DEVICE_MODELS_TABLE} WHERE device_type = adv.device_type) AS model_count,
+  (SELECT COUNT(*) FROM ${dbTables.DEVICE_VARIANT_TABLE} WHERE device_type = adv.device_type) AS variant_count
+  FROM ${dbTables.DEVICE_VERSION_TABLE} AS adv
```
Comment (naveen-repos) [inline src/controllers/device/devices.query.js:9]: Only get the count of the version's whose status is active.

File: src/controllers/device/devices.controller.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  errorResponse,
+  successResponse,
+} = require('../../utils/responseHandler');
+const { statusCodes } = require('../../utils/statusCode');
+const { successResponses } = require('./devices.constants');
+const { getDeviceTypes } = require('./devices.helper');
+
+exports.listDevicesTypes = async (req, res, next) => {
+  try {
+    const { success, errorCode, message, data } = await getDeviceTypes();
```
Comment (naveen-repos) [inline src/controllers/device/devices.controller.js:12]: const { success, errorCode, message, data = {} } = await getDeviceTypes();

please remove the return and 30. 

for this API the successful empty response should be 

```
{
   data: {
       device_types: [ ]
    }
}
```

File: src/controllers/device/devices.controller.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        message: successResponses.COUNT_FETCH_SUCCESSFULL.message,
+        code: statusCodes.STATUS_CODE_SUCCESS,
+      });
+    }
+    return successResponse({
+      res,
+      data: {},
+      message: successResponses.NO_DATA_FOUND.message,
+      code: statusCodes.STATUS_CODE_SUCCESS,
+    });
```
Comment (naveen-repos) [inline src/controllers/device/devices.controller.js:35]: Replace section with this
 
 return successResponse({
        res,
        data: {
          device_types: data,
        },
        message: successResponses.COUNT_FETCH_SUCCESSFULL.message,
        code: statusCodes.STATUS_CODE_SUCCESS,
      });

**Review by naveen-repos — CHANGES_REQUESTED** [review]

File: src/utils/statusCode.js (hunk 3 above)
Comment (satyanarayanaRaju21) [reply src/utils/statusCode.js:16]: anna these are common setupCode status codes

**Pull request merged**